scripts/aws.sh

```diff
@@ -0,0 +1,488 @@
+#!/bin/bash
+
+set -uo pipefail
+
+export AWS_PAGER=""
+
+err_log="err.log"
+
+trap 'rm -f "$err_log"' EXIT
+
+declare -A ami_map=(
+  [us-west-2,x86_64,rhel-9.2]=ami-0378fd0689802d015    # RHEL-9.2.0_HVM-20240229-x86_64-33-Hourly2-GP3
+  [us-west-2,x86_64,rhel-9.3]=ami-0c2f1f1137a85327e    # RHEL-9.3.0_HVM-20240229-x86_64-27-Hourly2-GP3
+  [us-west-2,x86_64,rhel-9.4]=ami-0423fca164888b941    # RHEL-9.4.0_HVM-20240605-x86_64-82-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.2]=ami-0cb125bb261a63f52     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.3]=ami-04379fa947a959c92     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.4]=ami-05b40ce1c0e236ef2     # RHEL-9.4.0_HVM-20240605-arm64-82-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.2]=ami-02c220fcee5dab581    # RHEL-9.2.0_HVM-20240229-x86_64-33-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.3]=ami-05463a02d11667441    # RHEL-9.3.0_HVM-20240229-x86_64-27-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.4]=ami-07d4917b6f95f5c2a    # RHEL-9.4.0_HVM-20240605-x86_64-82-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.2]=ami-06f9fb7baa169fdfc     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.3]=ami-016e6894567bb7f3e     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.4]=ami-02b8573b23fde21aa     # RHEL-9.4.0_HVM-20240605-arm64-82-Hourly2-GP3
+)
+
+spin() {
+  pid="${1}"
+  message_wait="${2}"
+  message_ok="${3}"
+
+  spin[0]="-"
+  spin[1]="\\"
+  spin[2]="|"
+  spin[3]="/"
+
+  while kill -0 "${pid}" 2>/dev/null
+  do
+    for i in "${spin[@]}"
+    do
+      echo -ne "\r${i} ${message_wait}"
+      sleep 0.1
+    done
+  done
+
+  wait "${pid}"
+  exit_status=$?
+
+  if [ "${exit_status}" -ne 0 ] ; then
+    cat "${err_log}"
+    exit 1
+  else
+    length=${#message_wait}
+    length=$((length + 2))
+    printf "\r%${length}s" ""
+    echo -e "\r${message_ok}"
+  fi
+
+}
+
+action_create() {
+  if [ -z "${inst_type}" ] ; then
+    echo "ERROR: inst_type not set" 1>&2
+    exit 1
+  fi
+
+  cf_tpl_file=cf-gen.yaml
+
+  EC2_AMI=""
+  MICROSHIFT_OS=rhel-9.4
+
+  ARCH="x86_64"
+  if [[ "${inst_type%.*}" =~ .*"g".* ]]; then
+    ARCH="arm64"
+  fi
+
+  if [[ "${EC2_AMI}" == "" ]]; then
+    EC2_AMI="${ami_map["${region},${ARCH},${MICROSHIFT_OS}"]}"
+  fi
+
+  ec2Type="VirtualMachine"
+  if [[ "${inst_type}" =~ c[0-9]+[gn].metal ]]; then
+    ec2Type="MetalMachine"
+  fi
+
+  ami_id=${EC2_AMI}
+
+  echo "Stack name: ${stack_name}"
+  echo "Instance type: ${inst_type}" 
+  echo "OS: ${MICROSHIFT_OS}" 
+  echo "ARCH: ${ARCH}" 
+  echo "AMI ID: ${EC2_AMI}" 
+  echo "region: ${region}" 
+
+  cat >"${cf_tpl_file}" <<EOF
```

Comment:
If we use the same template in CI, we should avoid copy-paste.

Reply:
I separated the file and I will make the CI script use that, as we discussed.